src/commands/infra/generate.ts

```diff
@@ -180,61 +195,63 @@ export const validateTemplateSources = async (
+    );
+  }
+  // Checking for git remote
+  const result = await simpleGit(sourcePath).listRemote([source]);
+  if (!result) {
+    logger.error(result);
+    throw new Error(
+      `Unable to clone the source remote repository. \
+The remote repo may not exist or you do not have the rights to access it`
+    );
+  }
+
+  logger.info(`Remote source repo: ${safeLoggingUrl} exists.`);
+  logger.info(
+    `Checking if source repo: ${safeLoggingUrl} has been already cloned to: ${sourcePath}.`
+  );
+
   try {
-    if (!fs.existsSync(sourcePath)) {
-      logger.warn(
-        `Provided source in template directory was not found, attempting to clone the template source repo locally.`
-      );
-      await createGenerated(sourcePath);
+    // Check if .git folder exists in ${sourcePath}, if not, then clone
+    // if already cloned, 'git pull'
+    if (fs.existsSync(path.join(sourcePath, ".git"))) {
+      // Make sure we have the latest version of all releases cached locally
+      await simpleGit(sourcePath).fetch("all");
+      await simpleGit(sourcePath).pull("origin", "master");
+      logger.info(`${safeLoggingUrl} already cloned. Performing 'git pull'...`);
     } else {
-      logger.info(
-        `Source template folder found. Validating existence of repository.`
-      );
-    }
-    // Checking for git remote
-    const result = await simpleGit(sourcePath).listRemote([source]);
-    if (!result) {
-      logger.error(
-        `Unable to clone the source remote repository. Does the remote repo exist? Do you have the rights to access it? ${result}`
-      );
-      return false;
-    } else {
-      logger.info(`Remote source repo: ${safeLoggingUrl} exists.`);
-      logger.info(
-        `Checking if source repo: ${safeLoggingUrl} has been already cloned to: ${sourcePath}.`
-      );
-      // Check if .git folder exists in ${sourcePath}, if not, then clone
-      // if already cloned, 'git pull'
-      if (fs.existsSync(path.join(sourcePath, ".git"))) {
-        // Make sure we have the latest version of all releases cached locally
-        await simpleGit(sourcePath).fetch("all");
-        await simpleGit(sourcePath).pull("origin", "master");
-        logger.info(
-          `${safeLoggingUrl} already cloned. Performing 'git pull'...`
-        );
-      } else {
-        await gitClone(source, sourcePath);
-      }
-      // Checkout tagged version
-      logger.info(`Checking out template version: ${version}`);
-      await simpleGit(sourcePath).checkout(version);
+      await gitClone(source, sourcePath);
     }
+    // Checkout tagged version
+    logger.info(`Checking out template version: ${version}`);
+
+    await simpleGit(sourcePath).checkout(version);
   } catch (err) {
-    logger.error(
-      `There was an error checking the remote source repository: ${err}`
-    );
-    return false;
+    logger.error(err);
+    // TOFIX: this error should be rethrown
+    // and if we do, infra-validations.sh will fail
```

Comment:
What do you mean by this error should be "rethrown"?

Reply:
if I do `throw err` so that the caller function can handle the exception and quit. then `infra-validations.sh` will not run to the end. This is what Nate mentioned about issue https://github.com/microsoft/bedrock/issues/893. We can `throw err` once this issue is address and `infra-validations.sh` can complete successfully.